### PR TITLE
qli chapter06

### PR DESCRIPTION
**Pull request by qianliok**

### CODE REVIEW

File: chapter06/qli/ex2.pl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+while( my $word = <STDIN> ) {
+   chomp $word; # having chomp inside while() gives uninitialized warnings for empty inputs
+   $words_hash{$word} += 1 ; 
+}
+
+print "#"x20 . " Summary sorted by points " . "#"x20 . "\n";
+foreach my $word (sort { $words_hash{$b} <=> $words_hash{$a} } keys %words_hash){
+  print "'$word' was seen $words_hash{$word} times \n";
+}
+
```
Comment (JPeace2) [inline chapter06/qli/ex2.pl:18]: I think we were meant to sort by the words not the number of times seen.

File: chapter06/qli/ex1.pl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    'barney' => 'rubble',
+    'wilma' => 'flintstone',
+    'empty' => '' 
+    );
+
+print "Please input a person's first name for their family name: ";
+chomp(my $name = <STDIN> );
+
+print exists $name_hash{$name} ? "$name\'s family name is $name_hash{$name} \n":"Unknown person \n";
+
```
Comment (JPeace2) [inline chapter06/qli/ex1.pl:17]: I think this line is a bit unclear. Perhaps you should add some brackets.

File: chapter06/qli/ex2.pl (hunk 1 above)
Comment (qianliok) [inline chapter06/qli/ex2.pl:18]: I thought the book asks to sort in code point order? But I noticed that not many people doing it, so maybe I misunderstood the English? 

> For extra credit, sort the summary words in code point order in the output.

Comment (JPeace2) [inline chapter06/qli/ex2.pl:18]: "Code point" order means the default ordering for strings, in other words Unicode order.

Comment (qianliok) [inline chapter06/qli/ex2.pl:18]: oh, I see. Thanks @JPeace2 :smiley:

**Pull request merged**